index.src.html

```diff
@@ -760,10 +760,10 @@ <h4 id="interfaces-credential-types-passwordcredential"><code>PasswordCredential
   :   <dfn attribute for="PasswordCredential">additionalData</dfn>
   ::  If the developer wishes to specify additional data to insert into the
       request body when submitting the credential information to a remote
-      endpoint, they can do so by assigning a {{FormData}} or
-      {{URLSearchParams}} object to this attribute. The credential information
-      will be mixed into the object to produce the body. The value is `null`
-      unless otherwise specified.
+      endpoint, they can do so by assigning a {{FormData}},
+      {{URLSearchParams}} or a generic object to this attribute. The
```

Comment:
As for nesting and arbitrary objects due to `any` -- if I read https://heycam.github.io/webidl/#es-record correctly, the conversion to IDL applies recursively. Sub-objects become sub-records. There are no setters left in the result.

(I'm afraid have no idea how to answer the first question about UA implementing the record.)